dsew_community_profile/delphi_dsew_community_profile/pull.py

```diff
@@ -543,8 +546,29 @@ def fetch_new_reports(params, logger=None):
         if SIGNALS[sig]["make_prop"]:
             ret[(geo, sig, IS_PROP)] = generate_prop_signal(df, geo, geomapper)
 
+    ret = interpolate_missing_values(ret)
+
     return ret
 
+def interpolate_missing_values(dfs: DataDict) -> DataDict:
+    interpolate_df = dict()
+    for key, df in dfs.items():
+        geo_dfs = []
+        for geo, group_df in df.groupby("geo_id"):
```

Comment:
how painful will this be for counties?

Reply:
on first impression, I don't think it will change at all. the only edge case that comes up is if certain counties are _really_ sparse, like to the point of only being reported selectively, when data is available (like fb survey). then things would get quite bad. not sure if that's a possibility in this indicator though?

Reply:
i meant more, will it take a long time to loop over 3600 counties x 7 signals?

Reply:
Ran a profile of the code on the first week of March data. Most of the indicator time is taken by reading the excel files, with the interpolation showing up at the bottom here (.5s total time on the scale of excel reading taking 10+s).

```py
         162936279 function calls (161659417 primitive calls) in 92.702 seconds

   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     8622    8.433    0.001    8.433    0.001 {method 'feed' of 'xml.etree.ElementTree.XMLParser' objects}
  4513240    5.941    0.000   13.615    0.000 _reader.py:177(parse_cell)
  4513240    4.569    0.000    5.190    0.000 _openpyxl.py:552(_convert_cell)
  4513240    3.941    0.000    4.305    0.000 cell.py:193(coordinate_to_tuple)
21827158/21826918    2.893    0.000    3.721    0.000 {built-in method builtins.isinstance}
  8916288    2.510    0.000    3.605    0.000 ElementTree.py:1292(read_events)
    34280    2.281    0.000    4.530    0.000 _read_only.py:101(_get_row)
    34312    2.037    0.000   31.770    0.001 _reader.py:113(parse)
  4513240    1.692    0.000    2.249    0.000 read_only.py:14(__init__)
  4040858    1.443    0.000    1.443    0.000 _reader.py:78(_cast_number)
8907712/8907673    1.259    0.000   13.695    0.000 ElementTree.py:1224(iterator)
    46905    1.046    0.000    3.345    0.000 managers.py:1061(iset)
  4513240    0.997    0.000    0.997    0.000 {method 'findtext' of 'xml.etree.ElementTree.Element' objects}
    34280    0.970    0.000   14.585    0.000 _reader.py:286(<listcomp>)
   279075    0.934    0.000    0.934    0.000 {method 'reduce' of 'numpy.ufunc' objects}
 13556161    0.931    0.000    0.931    0.000 {method 'get' of 'xml.etree.ElementTree.Element' objects}
526926/205994    0.803    0.000    5.524    0.000 {built-in method numpy.core._multiarray_umath.implement_array_function}
    34280    0.740    0.000    5.930    0.000 _openpyxl.py:576(<listcomp>)
3397777/2731437    0.725    0.000    0.942    0.000 {built-in method builtins.len}
4325556/4296437    0.661    0.000    0.857    0.000 {built-in method builtins.getattr}
   168610    0.622    0.000    1.118    0.000 generic.py:5525(__finalize__)
  8793866    0.621    0.000    0.621    0.000 read_only.py:108(value)
128896/25837    0.607    0.000    3.062    0.000 serialisable.py:46(from_tree)
  4513240    0.557    0.000    0.557    0.000 read_only.py:112(value)
        1    0.553    0.553   41.121   41.121 pull.py:553(interpolate_missing_values)
```

Reply:
Oh cool. It looks like we typically run in 1m15, so doubling that won't be a big deal day-to-day. Initial runs when adding new signals are 45m, so we'll need to be a bit careful when we add the test positivity signals, but that won't happen very often. 👍